server/consumer.go

```diff
@@ -1218,9 +1242,27 @@ func (o *Consumer) loopAndDeliverMsgs(s *Server, a *Account) {
 				o.mu.Lock()
 			}
 		}
+
 		// Track this regardless.
 		lts = ts
 
+		// If we have a rate limit set make sure we check that here.
+		if o.rlimit != nil {
+			now := time.Now()
+			r := o.rlimit.ReserveN(now, len(msg)+len(hdr)+len(subj)+len(dsubj)+len(o.ackReplyT))
+			delay := r.DelayFrom(now)
+			if delay > 0 {
+				qch := o.qch
```

Comment:
Is this channel closed when the consumer is closed? Does it happen when server is shutdown? If not, wait want to also select on s.quitCh to allow faster cleanup especially during tests.

Reply:
Yes it is closed when the consumer is stopped/deleted.